src/reducers/entities/channels.js

```diff
@@ -37,6 +37,17 @@ function currentChannelId(state = '', action) {
     }
 }
 
+function allChannels(state = {}, action) {
+    switch (action.type) {
+    case ChannelTypes.RECEIVED_ALL_CHANNELS:
+        return action.data;
```

Comment:
The `getAllChannels` action has paging, so if you were to page through a couple pages this would clobber previous results

Reply:
@jespino ☝️

Reply:
Yes, that is the intention here. Actually, we show the "last" getAllChannels or searchAllChannels results to show in the interface. And the idea is to not interfere with the normal channels used in other places (because if I wipe out that channels can generate problems in other places).

Reply:
Hmm if it's so ephemeral do we need to store it in redux at all? Would local state in the component be better?

Adding this reducer really adds a lot of potential for confusion. If I'm not familiar with exactly why this reducer exists how do I know when to look at `channels` versus`allChannels`? When should I use the `getAllChannels` action versus `getChannels`? How do I know the `allChannels` reducer is going to get purged before every new `getAllChannels` call, when none of our other actions follow that pattern?

Reply:
If it does need to be in redux, I think it needs to mesh better with our existing reducers and be clear as to what it's for

Reply:
Adding to this discussion, the intent of `channels` is to hold a list of information about any channel that the client knows about, not necessarily the user's current channels. Anything else that would store a channel just stores an ID to look up the entry in `channels` so that we aren't duplicating this data. Perhaps any of these other channels should be stored in `channels` and then we store the IDs that we're showing on that screen elsewhere (like in the webapp-specific `views` reducer)?

Reply:
Yes, I think you are right, I going to move this to the views, and try to reuse the channels.

Reply:
Done!